Build break on latest develop branch when ENABLE_LOG is OFF

### Bug Report
Build break on latest develop branch when ENABLE_LOG is OFF

##### Reproduction Steps
1. Try to build SDL with cmake ` -ENABLE_LOG=OFF` flag

##### Expected Behavior
SDL is built successfully

##### Observed Behavior
Build failed

##### OS & Version Information
* OS/Version: Ubuntu 16.04 amd64
* SDL Core Version: https://github.com/smartdevicelink/sdl_core/tree/develop (0186b8c)

##### Test Case, Sample Code, and / or Example App
- Console log
```
[  7%] Building CXX object src/components/protocol_handler/CMakeFiles/ProtocolHandler.dir/src/handshake_handler.cc.o
In file included from /home/db/ramdrv/sdl_core/src/components/include/utils/message_queue.h:39:0,
                 from /home/db/ramdrv/sdl_core/src/components/protocol_handler/include/protocol_handler/protocol_handler_impl.h:42,
                 from /home/db/ramdrv/sdl_core/src/components/protocol_handler/include/protocol_handler/handshake_handler.h:38,
                 from /home/db/ramdrv/sdl_core/src/components/protocol_handler/src/handshake_handler.cc:33:
/home/db/ramdrv/sdl_core/src/components/include/utils/conditional_variable.h:36:47: fatal error: boost/thread/condition_variable.hpp: No such file or directory
 #include <boost/thread/condition_variable.hpp>
```

--- a/src/components/protocol_handler/CMakeLists.txt
+++ b/src/components/protocol_handler/CMakeLists.txt
@@ -38,6 +38,7 @@ include_directories(
   ${CMAKE_BINARY_DIR}/src/components/
   ${LOG4CXX_INCLUDE_DIRECTORY}
   ${BSON_INCLUDE_DIRECTORY}
+  ${BOOST_INCLUDE_DIR}
 )
 
 set(PATHS

--- a/src/components/utils/include/utils/stl_utils.h
+++ b/src/components/utils/include/utils/stl_utils.h
@@ -32,6 +32,7 @@
 #ifndef SRC_COMPONENTS_UTILS_INCLUDE_UTILS_STL_UTILS_H_
 #define SRC_COMPONENTS_UTILS_INCLUDE_UTILS_STL_UTILS_H_
 
+#include <cstddef>
 #include "utils/macro.h"
 
 namespace utils {

--- a/tools/policy_table_validator/CMakeLists.txt
+++ b/tools/policy_table_validator/CMakeLists.txt
@@ -5,6 +5,7 @@ include_directories(
   ${CMAKE_SOURCE_DIR}/src/components/utils/include/
   ${COMPONENTS_DIR}/smart_objects/include/
   ${JSONCPP_INCLUDE_DIRECTORY}
+  ${BOOST_INCLUDE_DIR}
 )
 
 message(STATUS "Using ${EXTENDED_POLICY} policy mode")

